lib/user_comm_screener.rb

```diff
@@ -193,6 +232,20 @@ def load_preference_map
     UserCommPrefs.new(acting_user, resolved_user_communication_preferences)
   end
 
+  def actor_preferences
+    @actor_preferences ||= {
+      muting: @target_users.keys & actor_communication_preferences.filter_map do |pref|
+        pref.user_id if pref.preference_type == 'muted'
+      end,
+      ignoring: @target_users.keys & actor_communication_preferences.filter_map do |pref|
+        pref.user_id if pref.preference_type == 'ignored'
+      end,
+      disallowed_pms_from: @target_users.keys - actor_communication_preferences.filter_map do |pref|
+        pref.user_id if pref.preference_type == 'allowed_pm'
+      end
+    }
```

Comment:
```suggestion
    @actor_preferences ||= begin
      user_ids_by_preference_type = actor_communication_preferences.reduce({}) do |hash, pref|
        hash[pref.preference_type] ||= []
        hash[pref.preference_type] << pref.user_id
        hash
      end

      {
        muting: @target_users.keys & user_ids_by_preference_type['muted],
        ignoring: @target_users.keys & user_ids_by_preference_type['ignored],
        disallowed_pms_from: @target_users.keys - user_ids_by_preference_type['allowed_pm']
      }
    end
```

Minor suggestion to avoid multiple loops while still keeping the code readable.

Reply:
Ah nice that is better, and the `&` and `-` intersections will be removed from the other comment too :+1: